vktrace/src/vktrace_layer/vktrace_lib_pageguardcapture.cpp

```diff
@@ -25,13 +25,20 @@
 #include "vktrace_lib_pageguardcapture.h"
 #include "vktrace_lib_pageguard.h"
 
-#if defined(WIN32) //page guard solution for windows
-
 
 PageGuardCapture::PageGuardCapture()
 {
     EmptyChangedInfoArray.offset = 0;
     EmptyChangedInfoArray.length = 0;
+
+#if defined(PLATFORM_LINUX)
+    // Open the /proc/self/clear_refs file. We'll write to that file
+    // when we want to clear all the page dirty bits in /proc/self/pagemap.
+    clearRefsFd = open("/proc/self/clear_refs", O_WRONLY);
+    if (clearRefsFd < 0)
+        VKTRACE_FATAL_ERROR("Open of /proc/self/clear_refs failed.");
+#endif
```

Comment:
It would be good to call pageRefsDirtyClear during initialization.
Some older kernels (like android uses) implement /proc/self/pagemap, but use a different definition of the pagemap bits until the first writing of "4" to clear_refs.